.github/workflows/publish-docfx.yml

```diff
@@ -30,6 +30,8 @@ jobs:
     steps:
     - name: Checkout
       uses: actions/checkout@8ade135a41bc03ea155e62e844d188df1ea18608
+      with:
+        ref: main
```

Comment:
Why is this necessary? You already state this only runs on main anyways.

Reply:
I did this mainly so that the "Edit this page" button for docfx would point to the main branch, instead of the temporary PR branch blob that Actions/Checkout creates. This isn't a great solution, though, so I instead set an ENV variable here 5ac064479483e4279994905b7b4f436547799275